src/plugins/extra/typographer.rs

```diff
@@ -0,0 +1,91 @@
+//! Syntax highlighting for code blocks
+use crate::parser::core::CoreRule;
+use crate::parser::inline::Text;
+use crate::{MarkdownIt, Node};
+
+use lazy_static::lazy_static;
+use regex::{Regex, Replacer};
+
+use crate::plugins::extra::linkify::LinkifyMarker;
+lazy_static! {
+    static ref REPLACEMENTS: Box<[(Regex, &'static str)]> = Box::new([
+        (Regex::new(r"\+-").unwrap(), "±"),
+        (Regex::new(r"\.{2,}").unwrap(), "…"),
+        (Regex::new(r"([?!])…").unwrap(), "$1.."),
+        (Regex::new(r"([?!]){4,}").unwrap(), "$1$1$1"),
+        (Regex::new(r",{2,}").unwrap(), ","),
+        // Note: the regex docs state that multi-line regexes (with "(?m)" that
+        // is) are not unicode-aware, meaning they *only* match '\n' as the
+        // newline and not any other newline characters which... apparently
+        // exist... Oh well...
```

Comment:
I just checked this by inserting this test into `markdown-it-typographer.rs`:

```rust
#[test]
fn can_handle_unicode_next_line() {
    run("--\u{000B}---", "<p>– —</p>");
}
```

This uses the [line tabulation](https://www.fileformat.info/info/unicode/char/000b/index.htm) character which `println!` does interpret as a newline (but preserving the horizontal indentation as expected).

The typography works fine, it produces `"<p>–\u{b}—</p>\n"` and the same happens in the JS lib (I checked). Other newlines like `\u{0085}` aka [next line](https://www.fileformat.info/info/unicode/char/0085/index.htm) get completely ignored by JS, while the rust version would produce `\u{85}` in the output.

So for the typography I will actually remove the comment since it works fine (matching these more exotic newlines works just as fine as matching `$` or `^`) but I'm not sure what you mean by "other newlines will get normalized". Is that something that is planned, or a missing feature, or should it already work and it' somehow broken in these tests?